shell/browser/api/electron_api_web_contents.cc

```diff
@@ -741,10 +735,7 @@ WebContents::WebContents(v8::Isolate* isolate,
   // Whether to enable DevTools.
   options.Get("devTools", &enable_devtools_);
 
-  // BrowserViews are not attached to a window initially so they should start
-  // off as hidden. This is also important for compositor recycling. See:
-  // https://github.com/electron/electron/pull/21372
-  bool initially_shown = type_ != Type::kBrowserView;
+  bool initially_shown = true;
```

Comment:
I think there should be a test for this written. If a `WebContentsView` is created, but not set as a window's `contentView`, it should have `document.visibilityState === 'hidden'`. Once set as the contentView, it should be 'visible'.

Reply:
Done, + found a bug.